### DISCUSSION TITLE
Build Instructions?

### DISCUSSION THREAD

**Discussion by HotdogMassacre** [phonowell/genshin-impact-script#93] [Q&A]
Hi there, I am trying to get my own build of this so I can add some features. I am familiar with AHK but have never used NPM / Coffee-AHK sadly. So currently I am having a slight bit of trouble getting the import transpile thing to go through. Do you have any documentation on how to build the app so we can try to help add code fixes for you? I know you are likely quite busy.

Thank you @phonowell

**Comment by phonowell** [comment]
Sry for just seeing this now

Building the project itself is quite simple, you just need to clone the whole project and install it, then run `npm run build`
However, some of the files are missing because they are separated into other projects:
- Third-party libraries (e.g. gdip)
- Characters data

As for Coffee-AHK itself, this documentation may be helpful:
https://github.com/phonowell/coffee-ahk/blob/master/doc.md

**Comment by HotdogMassacre** [comment]
@phonowell  Yeah i think this was the problem i faced. I have Gdip and i was looking for the character data but it looks like you dont share those. Would you mind sharing them so that I can learn more about your coffee2ahk and also learn to add features for myself? I would be happy to share my code back to you in the event you wanted it.

**Comment by mmn6d6d6e** [comment]
@HotdogMassacre I found out that the missing dependencies existed at some point on the previous commit.

The character data removed on v0.0.31 (6ec799835471b54b661d081b7dbabf4416ff1c2b) and .ahk libraries removed on v0.0.32 (353056f0a4b4411a9dab13a81d7ee17546bb58e1).

You can checkout one commit before each of those commits and then copy the files to the appropriate path.

But of course, those files might not be the latest used to build the released executables on the release page.
For example, there are no character data for Ayato on those.

@phonowell Will you upload the character data & the third parties libraries as a separate project in the future? Or is there another way we can get those files?

**Reply by phonowell** [reply]
Sorry to reply only now due to busy work and the impact of the lockdown
Here are the related files

[lib.zip](https://github.com/phonowell/genshin-impact-script/files/8839713/lib.zip)
[source.zip](https://github.com/phonowell/genshin-impact-script/files/8839714/source.zip)

If you encounter problems in the building, contact me and I will reply asap
